homeassistant/components/climate/warmup4ie.py

```diff
@@ -0,0 +1,232 @@
+r"""
+platform that offers a connection to a warmup4ie device.
+
+this platform is inspired by the following code:
+https://github.com/alyc100/SmartThingsPublic/tree/master/devicetypes/alyc100/\
+warmup-4ie.src
+
+to setup this component, you need to register to warmup first.
+see
+https://my.warmup.com/login
+
+Then add to your
+configuration.yaml
+
+climate:
+  - platform: warmup4ie
+    name: YOUR_DESCRIPTION
+    username: YOUR_E_MAIL_ADDRESS
+    password: YOUR_PASSWORD
+    location: YOUR_LOCATION_NAME
+    room: YOUR_ROOM_NAME
+
+# the following issues are not yet implemented, since i have currently no need
+# for them
+# OPEN  - holiday mode still missing
+#       - commands for setting/retrieving programmed times missing
+"""
+
+import logging
+import voluptuous as vol
+from homeassistant.components.climate import ClimateDevice, PLATFORM_SCHEMA
+try:
+    from homeassistant.components.climate import (SUPPORT_TARGET_TEMPERATURE,
+                                                  SUPPORT_AWAY_MODE,
+                                                  SUPPORT_OPERATION_MODE,
+                                                  SUPPORT_ON_OFF, STATE_AUTO,
+                                                  STATE_MANUAL)
+except ImportError:
+    from homeassistant.components.climate.const import (
+        SUPPORT_TARGET_TEMPERATURE, SUPPORT_AWAY_MODE, SUPPORT_OPERATION_MODE,
+        SUPPORT_ON_OFF, STATE_AUTO, STATE_MANUAL)
+
+from homeassistant.const import (
+    TEMP_CELSIUS, ATTR_TEMPERATURE, CONF_NAME, CONF_USERNAME, CONF_PASSWORD)
+from homeassistant.exceptions import PlatformNotReady
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['warmup4ie==0.1.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_FLAGS = (SUPPORT_TARGET_TEMPERATURE | SUPPORT_AWAY_MODE |
+                 SUPPORT_OPERATION_MODE | SUPPORT_ON_OFF)
+
+CONF_LOCATION = 'location'
+CONF_ROOM = 'room'
```

Comment:
Import this from `const.py`.

Reply:
What exactly do you mean? CONF_LOCATION and CONF_ROOM? These values are not listed in const.py. Please clarify.

Reply:
Sorry, found it. CONF_ROOM is already defined in const.py. Fixed.